cloud/amazon/acm_certificate_facts.py

```diff
@@ -0,0 +1,504 @@
+            u'Status': u'ISSUED',
+            u'SubjectAlternativeNames': [
+                u'*.api.foo.com'
+            ],
+            u'DomainName': u'*.api.foo.com',
+            u'NotBefore': datetime.datetime(2016, 6, 2, 17, 0, tzinfo=tzlocal()),
+            u'NotAfter': datetime.datetime(2017, 7, 3, 5, 0, tzinfo=tzlocal()),
+            u'KeyAlgorithm': u'RSA-2048',
+            u'InUseBy': [],
+            u'SignatureAlgorithm': u'SHA256WITHRSA',
+            u'CreatedAt': datetime.datetime(2016, 6, 3, 7, 18, 18, tzinfo=tzlocal()),
+            u'IssuedAt': datetime.datetime(2016, 6, 3, 10, 32, 39, tzinfo=tzlocal()),
+            u'Serial': u'07:4b:97:96:e5:87:e4:2e:0d:ac:34:aa:3d:45:74:6f',
+            u'Issuer': u'Amazon',
+            u'DomainValidationOptions': [
+                {
+                    u'ValidationEmails': [
+                        u'webmaster@foo.com', u'admin@foo.com', u'administrator@foo.com', u'hostmaster@foo.com', u'postmaster@foo.com'
+                    ],
+                    u'ValidationDomain': u'foo.com',
+                    u'DomainName': u'*.api.foo.com'
+                }
+            ],
+            u'Subject': u'CN=*.api.foo.com'
+        }
+
+    Returns:
+        Tuple (bool, str, dict)
+    """
+    success = True
+    err_msg = ''
+    results = dict()
+    try:
+        if not check_mode:
+            results = client.describe_certificate(CertificateArn=arn)['Certificate']
+        else:
+            if arn == ARN:
+                results = DESCRIBE_CERTIFICATE
+            else:
+                account_id =  arn.split(':')[4]
+                BOTO_DESCRIBE_ERROR['Error']['Message'] = (
+                    'Could not find certificate {0} in account {1}'
+                    .format(arn, account_id)
+                )
+                raise(
+                    botocore.exceptions.ClientError(BOTO_DESCRIBE_ERROR, 'DescribeCertificate')
+                )
+
+    except Exception as e:
+        success = False
+        err_msg = e.response['Error']['Message']
+
+    return (success, err_msg, results)
+
+
+@ec2.AWSRetry.backoff()
+def list_certificates(client, check_mode=False):
+    """ Wrapper function for list_certificate
+    Args:
+        client (botocore.client.acm): The boto3 acm instance.
+
+    Basic Usage:
+        >>> client = boto3.client('acm', 'us-west-2')
+        >>> list_certificate(client)
+        [
+            {
+                u'CertificateArn': u'arn:aws:acm:us-west-2:123456789:certificate/25b4ad8a-1e24-4001-bcd0-e82fb3554cd7',
+                u'DomainName': u'*.api.foo.com'
+            }
+        ]
+
+    Returns:
+        Tuple (bool, str, list)
+    """
+    success = True
+    err_msg = ''
+    results = list()
+    try:
+        if not check_mode:
+            results = client.list_certificates()['CertificateSummaryList']
+        else:
+            results = LIST_CERTIFICATES
+    except Exception as e:
+        success = False
+        err_msg = str(e)
```

Comment:
I'd prefer if you also surface the traceback here. Just pass `traceback.format_exc(e)` as the named param `exception` to `fail_json`.